foxglove_bridge_base/src/parameter.cpp

```diff
@@ -38,6 +38,11 @@ Parameter::Parameter(const std::string& name, std::string value)
     , _type(ParameterType::PARAMETER_STRING)
     , _value(value) {}
 
+Parameter::Parameter(const std::string& name, std::string value, ParameterType type)
+    : _name(name)
+    , _type(type)
+    , _value(value) {}
```

Comment:
Should this constructor assert that the type is either string or byte array?

Alternatively, should we change the constructor to accept a `std::span<std::byte>` or similar and do the base64 conversion internally?

This page has some interesting advice, seems to suggest strong typing for this type of use case rather than tag dispatching (which I was considering recommending): https://www.fluentcpp.com/2018/04/27/tag-dispatching/

Reply:
Yes, I think we can improve this here. When I implemented this, I decided against doing the conversion internally but I don't remember what the reasoning was. I will see what I can do